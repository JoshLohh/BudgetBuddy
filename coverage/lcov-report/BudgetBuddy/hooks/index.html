
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for BudgetBuddy/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> BudgetBuddy/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">73.78% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>152/206</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.32% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>45/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.79% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>40/59</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.31% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>145/190</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="settlementUtils.ts"><a href="settlementUtils.ts.html">settlementUtils.ts</a></td>
	<td data-value="88.23" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.23" class="pct high">88.23%</td>
	<td data-value="51" class="abs high">45/51</td>
	<td data-value="52.63" class="pct medium">52.63%</td>
	<td data-value="19" class="abs medium">10/19</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="44" class="abs high">40/44</td>
	</tr>

<tr>
	<td class="file empty" data-value="useColorScheme.ts"><a href="useColorScheme.ts.html">useColorScheme.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file low" data-value="useColorScheme.web.ts"><a href="useColorScheme.web.ts.html">useColorScheme.web.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="useGroupDetails.ts"><a href="useGroupDetails.ts.html">useGroupDetails.ts</a></td>
	<td data-value="79.25" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.25" class="pct medium">79.25%</td>
	<td data-value="135" class="abs medium">107/135</td>
	<td data-value="61.4" class="pct medium">61.4%</td>
	<td data-value="57" class="abs medium">35/57</td>
	<td data-value="72.5" class="pct medium">72.5%</td>
	<td data-value="40" class="abs medium">29/40</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="126" class="abs high">105/126</td>
	</tr>

<tr>
	<td class="file low" data-value="useGroups.ts"><a href="useGroups.ts.html">useGroups.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

<tr>
	<td class="file low" data-value="useThemeColor.ts"><a href="useThemeColor.ts.html">useThemeColor.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	</tr>

<tr>
	<td class="file low" data-value="useUser.ts"><a href="useUser.ts.html">useUser.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-07-28T04:50:26.944Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    