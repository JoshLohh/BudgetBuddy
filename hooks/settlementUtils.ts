// settlementUtils.ts
export function calculateBalances(members, expenses) {
  const paid = {};
  members.forEach(id => { paid[id] = 0; });
  expenses.forEach(exp => {
    paid[exp.paidBy] += parseFloat(exp.amount) || 0;
  });
  const total = expenses.reduce((sum, exp) => sum + (parseFloat(exp.amount) || 0), 0);
  const share = total / members.length;
  const balances = {};
  members.forEach(id => {
    balances[id] = +(paid[id] - share).toFixed(2);
  });
  return balances;
}

export function calculateSettlements(balances) {
  const creditors = [];
  const debtors = [];
  Object.entries(balances).forEach(([id, balance]) => {
    if (balance > 0.009) creditors.push({ id, amount: balance });
    else if (balance < -0.009) debtors.push({ id, amount: -balance });
  });
  creditors.sort((a, b) => b.amount - a.amount);
  debtors.sort((a, b) => b.amount - a.amount);

  const settlements = [];
  let i = 0, j = 0;
  while (i < debtors.length && j < creditors.length) {
    const d = debtors[i], c = creditors[j];
    const amount = Math.min(d.amount, c.amount);
    settlements.push({ from: d.id, to: c.id, amount: +amount.toFixed(2) });
    d.amount -= amount;
    c.amount -= amount;
    if (d.amount < 0.01) i++;
    if (c.amount < 0.01) j++;
  }
  return settlements;
}
