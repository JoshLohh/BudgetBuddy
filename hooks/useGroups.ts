import { useContext } from 'react'
import { GroupsContext } from '@/contexts/GroupsContext'

export function useGroups() {
    const context = useContext(GroupsContext)


    if(!context) {
        throw new Error("useUser must be used within a GroupProvider")
    }

    
    return context
}