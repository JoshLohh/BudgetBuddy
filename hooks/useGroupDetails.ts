// useGroupDetails.ts
import { databases } from '@/lib/appwrite';
import { Query } from 'appwrite';
import { useFocusEffect } from 'expo-router';
import { useCallback, useEffect, useState } from 'react';
import { calculateBalances, calculateSettlements } from './settlementUtils';

const databaseId = process.env.EXPO_PUBLIC_APPWRITE_DATABASE_ID ?? '';
const groupsCollectionId = process.env.EXPO_PUBLIC_APPWRITE_GROUPS_COLLECTION_ID ?? '';
const usersCollectionId = process.env.EXPO_PUBLIC_APPWRITE_USERS_COLLECTION_ID ?? '';
const expensesCollectionId = process.env.EXPO_PUBLIC_APPWRITE_EXPENSES_COLLECTION_ID ?? '';

export function useGroupDetails(groupId) {
  const [group, setGroup] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');
  const [membersExpanded, setMembersExpanded] = useState(false);
  const [memberProfiles, setMemberProfiles] = useState([]);
  const [searchModalVisible, setSearchModalVisible] = useState(false);
  const [searchQuery, setSearchQuery] = useState('');
  const [searchResults, setSearchResults] = useState([]);
  const [searching, setSearching] = useState(false);
  const [expenses, setExpenses] = useState([]);
  const [expensesLoading, setExpensesLoading] = useState(false);
  const [showAllExpenses, setShowAllExpenses] = useState(false);

  const EXPENSES_PREVIEW_COUNT = 10;

  // Fetch group info
  useEffect(() => {
    if (!groupId) return;
    setLoading(true);
    setError('');
    const id = Array.isArray(groupId) ? groupId[0] : groupId;
    databases
      .getDocument(databaseId, groupsCollectionId, id)
      .then(doc => {
        setGroup({
          id: doc.$id,
          title: doc.title,
          description: doc.description,
          members: doc.members ?? [],
          createdBy: doc.createdBy,
        });
        setLoading(false);
      })
      .catch(() => {
        setError('Group not found.');
        setLoading(false);
      });
  }, [groupId]);

  // Fetch expenses
  useEffect(() => {
    if (!groupId) return;
    setExpensesLoading(true);
    databases
      .listDocuments(
        databaseId,
        expensesCollectionId,
        [Query.equal('groupId', groupId)]
      )
      .then(res => {
        setExpenses(res.documents);
        setExpensesLoading(false);
      })
      .catch(() => setExpensesLoading(false));
  }, [groupId]);

  // Refetch expenses on focus
  useFocusEffect(
    useCallback(() => {
      setExpensesLoading(true);
      databases.listDocuments(
        databaseId,
        expensesCollectionId,
        [Query.equal('groupId', groupId)]
      ).then(res => {
        setExpenses(res.documents);
        setExpensesLoading(false);
      }).catch(() => setExpensesLoading(false));
    }, [groupId])
  );

  // Fetch member profiles
  useEffect(() => {
    if (!group || !group.members.length) {
      setMemberProfiles([]);
      return;
    }
    setLoading(true);
    Promise.all(
      group.members.map(userId =>
        databases
          .getDocument(databaseId, usersCollectionId, userId)
          .then(profile => ({ userId, username: profile.username }))
          .catch(() => ({ userId, username: '(unknown)' }))
      )
    ).then(profiles => {
      setMemberProfiles(profiles);
      setLoading(false);
    });
  }, [group]);

  // Helper to get username
  const getUsername = (userId) => {
    const profile = memberProfiles.find(p => p.userId === userId);
    return profile ? profile.username : userId;
  };

  // Search for users by username
  const handleSearch = async () => {
    setSearching(true);
    try {
      const res = await databases.listDocuments(
        databaseId,
        usersCollectionId,
        []
      );
      const filtered = res.documents.filter(doc =>
        doc.username?.toLowerCase().includes(searchQuery.trim().toLowerCase())
      );
      setSearchResults(filtered);
    } catch (e) {
      setSearchResults([]);
    }
    setSearching(false);
  };

  // Add member to group
  const handleAddMember = async userId => {
    if (!group) return;
    if (group.members.includes(userId)) return;
    const updatedMembers = [...group.members, userId];
    await databases.updateDocument(
      databaseId,
      groupsCollectionId,
      group.id,
      { members: updatedMembers }
    );
    setGroup({ ...group, members: updatedMembers });
    setSearchModalVisible(false);
    setSearchQuery('');
    setSearchResults([]);
  };

  // Remove member from group
  const handleRemoveMember = async userId => {
    if (!group) return;
    const updatedMembers = group.members.filter(id => id !== userId);
    await databases.updateDocument(
      databaseId,
      groupsCollectionId,
      group.id,
      { members: updatedMembers }
    );
    setGroup({ ...group, members: updatedMembers });
  };

  // Expenses preview logic
  const expensesToShow = showAllExpenses ? expenses : expenses.slice(0, EXPENSES_PREVIEW_COUNT);
  const hasMoreExpenses = expenses.length > EXPENSES_PREVIEW_COUNT;

  // Settlements calculation
  const balances = group ? calculateBalances(group.members, expenses) : {};
  const settlements = calculateSettlements(balances);

  // Total expenses
  const totalExpenses = expenses.reduce((sum, exp) => sum + (parseFloat(exp.amount) || 0), 0);

  return {
    group,
    loading,
    error,
    membersExpanded,
    setMembersExpanded,
    memberProfiles,
    searchModalVisible,
    setSearchModalVisible,
    searchQuery,
    setSearchQuery,
    searchResults,
    handleSearch,
    searching,
    handleAddMember,
    handleRemoveMember,
    expenses,
    expensesLoading,
    expensesToShow,
    hasMoreExpenses,
    showAllExpenses,
    setShowAllExpenses,
    settlements,
    getUsername,
    totalExpenses,
  };
}
