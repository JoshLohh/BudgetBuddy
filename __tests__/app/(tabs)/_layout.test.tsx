import React from 'react';
import { render } from '@testing-library/react-native';
import TabLayout from '../../../app/(tabs)/_layout';
import { useColorScheme } from '@/hooks/useColorScheme';
import { Platform, Text } from 'react-native';

jest.mock('@/hooks/useColorScheme');

jest.mock('expo-router', () => {
    const React = require('react');
    const { Text, View } = require('react-native');
  
    const Tabs = jest.fn(({ children, screenOptions }) => {
      const TabBarButton = screenOptions?.tabBarButton ?? (() => null);
      const TabBarBackground = screenOptions?.tabBarBackground ?? (() => null);
  
      return (
        <>
          {children}
          <TabBarButton testID="mock-haptic-tab" />
          <TabBarBackground testID="mock-tab-bar-background" />
        </>
      );
    }) as any;
  
    Tabs.Screen = (props: any) => {
      // Render title and invoke tabBarIcon prop which renders IconSymbol mock
      return (
        <>
          <Text testID={props.name}>{props?.options?.title || 'No Title'}</Text>
          {props?.options?.tabBarIcon && (
            <View testID="mock-icon-container">
              {props.options.tabBarIcon({ color: 'black' })}
            </View>
          )}
        </>
      );
    };
  
    return { Tabs };
});
  

jest.mock('@/components/auth/UserOnly', () => ({
  __esModule: true,
  default: ({ children }: any) => <>{children}</>,
}));

jest.mock('@/components/HapticTab', () => ({
  __esModule: true,
  HapticTab: (props: any) => {
    const { View } = require('react-native');
    return <View testID="mock-haptic-tab" {...props} />;
  },
}));

jest.mock('@/components/ui/IconSymbol', () => ({
  __esModule: true,
  IconSymbol: (props: any) => {
    const { View } = require('react-native');
    return <View testID="mock-icon" {...props} />;
  },
}));

jest.mock('@/components/ui/TabBarBackground', () => ({
  __esModule: true,
  default: (props: any) => {
    const { View } = require('react-native');
    return <View testID="mock-tab-bar-background" {...props} />;
  },
}));

jest.mock('@/constants/Colors', () => ({
  Colors: {
    light: { tint: '#123', background: '#abc' },
    dark: { tint: '#456', background: '#def' },
  },
}));

describe('TabLayout', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders all Tabs.Screen with correct titles', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getByText } = render(<TabLayout />);
    expect(getByText('Groups')).toBeTruthy();
    expect(getByText('Create')).toBeTruthy();
    expect(getByText('Profile')).toBeTruthy();
  });

  it('renders custom tab bar background and button', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getAllByTestId } = render(<TabLayout />);
    expect(getAllByTestId('mock-haptic-tab').length).toBeGreaterThan(0);
    expect(getAllByTestId('mock-tab-bar-background').length).toBeGreaterThan(0);
  });

  it('passes correct color scheme to Tabs (light)', () => {
    Object.defineProperty(Platform, 'OS', { value: 'android', configurable: true });
    const { Tabs } = require('expo-router');
    (useColorScheme as jest.Mock).mockReturnValue('light');
    render(<TabLayout />);
    const lastScreenOptions = Tabs.mock.calls[Tabs.mock.calls.length - 1][0]?.screenOptions;
    expect(lastScreenOptions.tabBarActiveTintColor).toEqual('#123');
    //expect(lastScreenOptions.tabBarStyle.backgroundColor).toEqual('#abc');
  });

  it('passes correct color scheme to Tabs (dark)', () => {
    Object.defineProperty(Platform, 'OS', { value: 'android', configurable: true });
    const { Tabs } = require('expo-router');
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    render(<TabLayout />);
    const lastScreenOptions = Tabs.mock.calls[Tabs.mock.calls.length - 1][0]?.screenOptions;
    expect(lastScreenOptions.tabBarActiveTintColor).toEqual('#456');
    //expect(lastScreenOptions.tabBarStyle.backgroundColor).toEqual('#def');
  });

  it('renders icons on tabs', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getAllByTestId } = render(<TabLayout />);
    expect(getAllByTestId('mock-icon').length).toBeGreaterThan(0);
  });

  it('applies iOS-specific tabBarStyle', () => {
    Object.defineProperty(Platform, 'OS', { value: 'ios', configurable: true });
    const { Tabs } = require('expo-router');
    (useColorScheme as jest.Mock).mockReturnValue('light');
    render(<TabLayout />);
    const lastScreenOptions = Tabs.mock.calls[Tabs.mock.calls.length - 1][0]?.screenOptions;
    expect(lastScreenOptions.tabBarStyle.position).toBe('absolute');
  });

  it('applies android-specific tabBarStyle', () => {
    Object.defineProperty(Platform, 'OS', { value: 'android', configurable: true });
    
    // Mock Platform.select to behave properly
    jest.spyOn(Platform, 'select').mockImplementation((obj: any) => {
        return obj.android ?? (obj as any).default ?? {};
    });
  
    const { Tabs } = require('expo-router');
    (useColorScheme as jest.Mock).mockReturnValue('light');
    render(<TabLayout />);
    const lastScreenOptions = Tabs.mock.calls[Tabs.mock.calls.length - 1][0]?.screenOptions;
  
    expect(lastScreenOptions.tabBarStyle).toEqual({});
  });

  it('handles null color scheme gracefully', () => {
    const { Tabs } = require('expo-router');
    (useColorScheme as jest.Mock).mockReturnValue(null);
    render(<TabLayout />);
    const lastScreenOptions = Tabs.mock.calls[Tabs.mock.calls.length - 1][0]?.screenOptions;
    expect(lastScreenOptions.tabBarActiveTintColor).toEqual('#123'); // fallback to light theme
  });

  it('applies light mode correctly (snapshot)', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { toJSON } = render(<TabLayout />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies dark mode correctly (snapshot)', () => {
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    const { toJSON } = render(<TabLayout />);
    expect(toJSON()).toMatchSnapshot();
  });

  /*it('does not render tabs if unauthenticated', () => {
    jest.resetModules();
    jest.doMock('@/components/auth/UserOnly', () => ({
      __esModule: true,
      default: () => null, // simulating unauthenticated user hides content
    }));
    const { getByTestId } = require('@testing-library/react-native');
    const TabLayoutReloaded = require('../../../app/(tabs)/_layout').default;
    const tree = getByTestId ? render(<TabLayoutReloaded />).toJSON() : null;
    expect(tree).toBeNull();
  });*/
});



describe('TabLayout (unauthenticated)', () => {
    let TabLayoutReloaded: React.ComponentType;
    beforeAll(() => {
        jest.resetModules();
        jest.doMock('@/components/auth/UserOnly', () => ({
          __esModule: true,
          default: () => null,
        }));
        TabLayoutReloaded = require('../../../app/(tabs)/_layout').default; // load after mock
      });
      
      afterAll(() => {
        jest.resetModules();
      });
      
      it('does not render tabs if unauthenticated', () => {
        const tree = render(<TabLayoutReloaded />).toJSON();
        expect(tree).toBeNull();
      });
});
