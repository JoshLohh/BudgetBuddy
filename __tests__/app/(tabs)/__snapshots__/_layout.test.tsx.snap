// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabLayout applies dark mode correctly (snapshot) 1`] = `
[
  <Text
    testID="group"
  >
    Groups
  </Text>,
  <View
    testID="mock-icon-container"
  >
    <View
      color="black"
      name="person.3.fill"
      size={38}
      testID="mock-icon"
    />
  </View>,
  <Text
    testID="create"
  >
    Create
  </Text>,
  <View
    testID="mock-icon-container"
  >
    <View
      color="black"
      name="plus.square.fill.on.square.fill"
      size={28}
      testID="mock-icon"
    />
  </View>,
  <Text
    testID="profile"
  >
    Profile
  </Text>,
  <View
    testID="mock-icon-container"
  >
    <View
      color="black"
      name="person.fill"
      size={28}
      testID="mock-icon"
    />
  </View>,
  <View
    testID="mock-haptic-tab"
  />,
  <View
    testID="mock-tab-bar-background"
  />,
]
`;

exports[`TabLayout applies light mode correctly (snapshot) 1`] = `
[
  <Text
    testID="group"
  >
    Groups
  </Text>,
  <View
    testID="mock-icon-container"
  >
    <View
      color="black"
      name="person.3.fill"
      size={38}
      testID="mock-icon"
    />
  </View>,
  <Text
    testID="create"
  >
    Create
  </Text>,
  <View
    testID="mock-icon-container"
  >
    <View
      color="black"
      name="plus.square.fill.on.square.fill"
      size={28}
      testID="mock-icon"
    />
  </View>,
  <Text
    testID="profile"
  >
    Profile
  </Text>,
  <View
    testID="mock-icon-container"
  >
    <View
      color="black"
      name="person.fill"
      size={28}
      testID="mock-icon"
    />
  </View>,
  <View
    testID="mock-haptic-tab"
  />,
  <View
    testID="mock-tab-bar-background"
  />,
]
`;
